pkg/metrics/metrics.go

```diff
@@ -86,6 +91,42 @@ func (m *Registry) GetGauge(key string) metrics.Gauge {
 	return m.createGauge(combinedKey)
 }
 
+// GetTimer gets go-kit expvar Counter
+func (m *Registry) GetTimer(key string) go_kit_metrics.Counter {
+	if key == "" {
+		log.Warn().Msg("metrics timer key is empty")
+		return nil
+	}
+
+	combinedKey := TimerPrefix + "." + key
+
+	m.counterLock.Lock()
+	defer m.counterLock.Unlock()
+	if val, ok := m.metricsCounterVars[combinedKey]; ok {
+		return val
+	}
+
+	return m.createCounter(combinedKey)
+}
```

Comment:
We should define `Timer` on the `MetricsRegistry` as an abstract construct and not just an alias for a `Counter`. This moves this logic out of this middleware implementation which has the benefit of being reusable directly out of the registry and removes the go-kit dependency from that module.
```go
type Timer struct {
	hits  go_kit_metrics.Counter
	total go_kit_metrics.Counter
	hist  go_kit_metrics.Histogram
}

func (m *Registry) NewTimer(key string) *Timer {
	...
	name := TimerPrefix + "." + key
	return &Timer {
		hits:  m.createCounter(name + ".hits"),
		total: m.createCounter(name + ".total"),
		hist:  m.createHistogram(name+".hist"),
	}
}

func (t *Timer) Update(delta float64) {
	t.hitCounts.Add(1)
	t.responseTime.Add(timeDiff)
	t.responseTimeHist()
}
```

Reply:
I have made this change, I like my previous implementation - I don't need to make a map specially for timers. Take a look...